File: solutions/javascript/collatz-conjecture/2/collatz-conjecture.js
/**
* Calculates the number of steps it takes a certain number to reach 1 as per the Collatz Conjecture
*
* @param {number} n A candidate to be reduce to 1 using Collatz.
* @returns {number} The number of steps it takes to reach 1 using Collatz.
*/
export const steps = (n) => {
  if (n < 1) throw new Error("Only positive numbers are allowed");
  for (var steps = 0; n != 1; steps++) n % 2 == 0 ? n /= 2 : n = 3 * n + 1;
  return steps;
};

File: solutions/python/collatz-conjecture/2/collatz_conjecture.py
def steps(number):
    """It takes in a number and the applies the Collatz conjecture calculating how many steps it takes to get to one
    :param number: The number to be used for the conjecture.
    :return: int - The amount of steps it takes to reduce to one.
    """

    if number <= 0: raise ValueError("Only positive integers are allowed")

    steps = 0
    while number != 1:
        steps += 1
        number = number / 2 if number % 2 == 0 else number * 3 + 1

    return steps
